CMakePresets.json

```diff
@@ -0,0 +1,307 @@
+{
+    "version": 3,
+    "configurePresets": [
+        {
+            "hidden": true,
+            "name": "linux",
+            "condition": {
+                "type": "equals",
+                "lhs": "${hostSystemName}",
+                "rhs": "Linux"
+            }
+        },
+        {
+            "hidden": true,
+            "name": "ci-base",
+            "displayName": "CI configuration",
+            "description": "Sets Ninja generator, build and install directory",
+            "generator": "Ninja",
+            "binaryDir": "${sourceDir}/out/build/${presetName}",
+            "installDir": "${sourceDir}/out/install/${presetName}",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Release",
+                "BLA_VENDOR": "Generic"
+            },
+            "warnings": {
+                "dev": true,
+                "deprecated": true
+            }
+        },
+        {
+            "name": "default",
+            "displayName": "Default build configuration",
+            "description": "The right build configuration for most users",
+            "inherits": "ci-base",
+            "cacheVariables": {
+                "BUILD_TESTING": false,
+                "CMAKE_COMPILE_WARNING_AS_ERROR": false
+            },
+            "warnings": {
+                "dev": false,
+                "deprecated": false
+            }
+        },
+        {
+            "name": "dev-default",
+            "displayName": "Default build configuration for devs",
+            "description": "The right build configuration for most developers",
+            "inherits": "ci-base",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "RelWithDebInfo",
+                "CMAKE_CXX_FLAGS": "-Wall"
```

Comment:
`-Wall` for dev-default may be too opinionated. Happy to remove/change, but it did highlight a number of potential issues (unused variables, etc)